src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -1645,42 +1649,47 @@ void ScriptEngine::UpdateIntervals()
     {
         // timestamp has wrapped, subtract all intervals by the remaining amount before wrap
         auto delta = static_cast<int64_t>(std::numeric_limits<uint32_t>::max() - _lastIntervalTimestamp);
-        for (auto& interval : _intervals)
+        for (auto& entry : _intervals)
         {
-            if (interval.IsValid())
-            {
-                interval.LastTimestamp = -delta;
-            }
+            auto& interval = entry.second;
+            interval.LastTimestamp = -delta;
         }
     }
     _lastIntervalTimestamp = timestamp;
 
-    for (auto& interval : _intervals)
+    // This loop needs to account for removal and insertions during iteration.
+    for (auto it = _intervals.begin(), itNext = it; it != _intervals.end(); it = itNext)
     {
-        if (interval.IsValid())
+        itNext++;
+
+        auto& interval = it->second;
+
+        if (timestamp >= interval.LastTimestamp + interval.Delay)
         {
-            if (timestamp >= interval.LastTimestamp + interval.Delay)
-            {
-                ExecutePluginCall(interval.Owner, interval.Callback, {}, false);
+            ExecutePluginCall(interval.Owner, interval.Callback, {}, false);
 
-                interval.LastTimestamp = timestamp;
-                if (!interval.Repeat)
-                {
-                    RemoveInterval(nullptr, interval.Handle);
-                }
+            interval.LastTimestamp = timestamp;
+            if (!interval.Repeat)
+            {
+                _intervals.erase(it);
             }
         }
     }
 }
 
 void ScriptEngine::RemoveIntervals(const std::shared_ptr<Plugin>& plugin)
 {
-    for (auto& interval : _intervals)
+    for (auto it = _intervals.begin(); it != _intervals.end();)
     {
+        auto& interval = it->second;
+
         if (interval.Owner == plugin)
         {
-            interval = {};
+            it = _intervals.erase(it);
+            continue;
         }
+
+        it++;
```

Comment:
```suggestion
            continue;
        }
        else
        {
            it++;
        }
```
Would this be easier to understand? Not sure, but it's how I added the suggestion above.

Reply:
I'm not torn about this, a habit of mine to just bail as early as possible, it won't make a difference code-gen wise. Changed it.